chain/jsonrpc-primitives/src/errors.rs

```diff
@@ -14,12 +14,33 @@ pub struct RpcParseError(pub String);
 #[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
 #[serde(deny_unknown_fields)]
 pub struct RpcError {
+    #[serde(flatten)]
+    pub error_struct: Option<RpcErrorKind>,
+    /// Deprecated please use the `error_struct` instead
     pub code: i64,
+    /// Deprecated please use the `error_struct` instead
     pub message: String,
+    /// Deprecated please use the `error_struct` instead
     #[serde(skip_serializing_if = "Option::is_none")]
     pub data: Option<Value>,
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+#[serde(tag = "name", content = "cause", rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum RpcErrorKind {
+    ValidationError(ValidationErrorKind),
```

Comment:
I suggest we consider the following naming to disambiguate the "validation" term.

```suggestion
    RequestValidationError(RpcRequestValidationErrorKind),
```